dlr/dlr-engine/src/main/java/ai/djl/dlr/engine/DlrPredictor.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.dlr.engine;
+
+import ai.djl.Device;
+import ai.djl.dlr.jni.JniUtils;
+import ai.djl.inference.Predictor;
+import ai.djl.translate.Translator;
+
+/**
+ * {@code DlrPredictor} is special implementation of {@link Predictor} for DLR.
+ *
+ * <p>The native Dlr doesn't support multi-threading feature, when creating a new DlrPredictor, we
```

Comment:
Can this be done in the DlrModel? So, the model would track whether it is the first created predictor in the .newPredictor(...), and if not give it a new handle.

Reply:
Yes we can, but we need to override the getBlock to do implicitly copy, which is not ideal for end user. Let's say if user call DlrModel.getBlock() to play around the object they don't realize we reload another new TVM model.